src/Console/Command/Hook.php

```diff
@@ -55,6 +56,50 @@ protected function configure(): void
             InputOption::VALUE_OPTIONAL,
             'Relative path from your config file to your bootstrap file'
         );
+
+        $this->addOption(
+            'list-actions',
+            'l',
+            InputOption::VALUE_NONE,
+            'List actions for this hook without running the hook'
+        );
+
+        $this->addOption(
+            'action',
+            'a',
+            InputOption::VALUE_IS_ARRAY | InputOption::VALUE_REQUIRED,
+            'Run only the actions listed'
+        );
+
+        $this->addOption(
+            'disable-plugins',
+            null,
+            InputOption::VALUE_NONE,
+            'Disable all hook plugins'
+        );
+    }
+
+    /**
+     * Initialize the command by checking/modifying inputs before validation
+     *
+     * @param  \Symfony\Component\Console\Input\InputInterface   $input
+     * @param  \Symfony\Component\Console\Output\OutputInterface $output
+     * @return void
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        // If `--list-actions` is present, we will ignore any arguments, since
+        // this option intends to output a list of actions for the hook without
+        // running the hook. So, if any arguments are required but not present
+        // in the input, we will set them to an empty string in the input to
+        // suppress any validation errors.
+        if ($input->getOption('list-actions') === true) {
+            foreach ($this->getDefinition()->getArguments() as $arg) {
+                if ($arg->isRequired() && $input->getArgument($arg->getName()) === null) {
+                    $input->setArgument($arg->getName(), '');
+                }
+            }
+        }
```

Comment:
What do you think about creating three separate commands (e.g. `hook:pre-commit:list-actions`) instead of adding those as options with need for this magic?

Reply:
`captainhook list-actions pre-commit`

Reply:
@ramsey have you seen this comment as well?

Reply:
This sounds good to me.